nifi-marklogic-processors/src/main/java/org/apache/nifi/marklogic/processor/ExtensionCallMarkLogic.java

```diff
@@ -188,9 +173,13 @@ protected void handleExtensionCallResults(ServiceResultIterator results, Process
         }
         try {
             while (results.hasNext()) {
-                session.append(originalFlowFile, out -> out.write(results.next().getContent(new BytesHandle()).get()));
+                ResourceServices.ServiceResult file = results.next();
```

Comment:
Forgot to ask - did you run into an issue where a `ServiceResult` is returned but it has no length, and thus it shouldn't be written?

Reply:
yes, I did, hence I added the if below file.getLength() >0